### PR TITLE
update fn setup

### PR DESCRIPTION
**Pull request by davidiw**
* docs pointed to diem and links were broken
* seed_addrs was broken or_default + extend is not ideal
* Upstream from config + ValidatorFullNode from onchain results in no
  config

### CODE REVIEW

File: network/builder/src/builder.rs [hunk 1]
```diff
@@ -238,11 +238,13 @@ impl NetworkBuilder {
                 .seed_addrs
                 .iter()
                 .map(|(peer_id, addrs)| {
-                    (peer_id, Peer::from_addrs(PeerRole::Upstream, addrs.clone()))
+                    (peer_id, Peer::from_addrs(PeerRole::ValidatorFullNode, addrs.clone()))
```
Comment (gregnazario) [inline network/builder/src/builder.rs:241]: Well, technically this doesn't always have to be true, but it should be fine given the backwards compatibility (and eventual removal)

File: network/builder/src/builder.rs [hunk 2]
```diff
@@ -238,11 +238,13 @@ impl NetworkBuilder {
...
                 })
                 .for_each(|(peer_id, peer)| {
-                    let seed = seeds.entry(*peer_id).or_default();
-                    seed.extend(peer).unwrap();
+                    seeds.entry(*peer_id)
+                        // Sad clone due to Rust not realizing these are two distinct paths
+                        .and_modify(|seed| seed.extend(peer.clone()).unwrap())
+                        .or_insert(peer);
```
Comment (gregnazario) [inline network/builder/src/builder.rs:247]: Oh, sad

**Review by gregnazario — APPROVED** [review]

**Comment by davidiw** [comment]
I don't think I write very good rust code, so apparently lints failed :trollface:

**Comment by davidiw** [comment]
Well this is frustrating... for some reason it cannot build "state-sync" and there's no clear error message... let's try again

/land

**Pull request closed without merging**